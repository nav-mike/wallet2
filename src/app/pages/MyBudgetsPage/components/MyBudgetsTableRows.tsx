import React from 'react';
import { IBudget, IMyBudgetsTableProps } from './MyBudgetsTable';
import { MyBudgetsTableItem } from './MyBudgetsTableItem';

export const MyBudgetsTableRows = (props: IMyBudgetsTableProps) => {
  const data = props.budgets.map((item: IBudget) => (
    <MyBudgetsTableItem item={item} key={`${item.month}-${item.year}`} />
  ));
  return <tbody>{data}</tbody>;
};
