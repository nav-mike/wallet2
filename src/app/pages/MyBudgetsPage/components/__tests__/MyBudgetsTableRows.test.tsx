import * as React from 'react';
import { render } from '@testing-library/react';
import { MyBudgetsTableRows } from '../MyBudgetsTableRows';

describe('<MyBudgetsTableRows />', () => {
  const data = [
    {
      month: 'January',
      year: 2021,
      value: 500,
    },
    {
      month: 'February',
      year: 2021,
      value: 500,
    },
  ];

  it('should render and match the snapshot', () => {
    const rows = render(<MyBudgetsTableRows budgets={data} />, {
      container: document.createElement('table'),
    });
    expect(rows.container.firstChild).toMatchSnapshot();
  });
});
