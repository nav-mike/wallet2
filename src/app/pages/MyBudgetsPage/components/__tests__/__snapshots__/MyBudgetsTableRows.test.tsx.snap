// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyBudgetsTableRows /> should render and match the snapshot 1`] = `
<tbody>
  <tr>
    <td>
      2021
    </td>
    <td>
      January
    </td>
    <td>
      500
    </td>
  </tr>
  <tr>
    <td>
      2021
    </td>
    <td>
      February
    </td>
    <td>
      500
    </td>
  </tr>
</tbody>
`;
